### ISSUE TITLE
Double-clicking progress bar should scroll pagemark to top of window

### ISSUE THREAD

**Issue by sopoforic** [burtonator/polar-bookshelf#1544]
2.0.56 on Windows 10

Double-clicking the progress bar on a partly-read PDF scrolls the furthest pagemark into view. However, it scrolls so that the bottom of the last pagemark is at the bottom of the window, which means additional scrolling is always required prior to reading.

If the zoom level is such that a full page fits on screen, it would be sensible to position the last partly-completed page at the top of the window, or the first unread page if the previous page was fully read. Otherwise, the bottom of the pagemark should be near (but not quite at) the top of the screen, so a little context is visible, as well as mostly unread text.

**Comment by burtonator** [comment]
Agreed. I'll work on a fix for this shortly. I added it to the top of the queue.  It's a small fix to nail this but I just needed to refactor it a bit more.
